subprojects/core/src/integTest/groovy/org/gradle/internal/operations/logging/LoggingBuildOperationProgressIntegTest.groovy

```diff
@@ -427,14 +428,18 @@ class LoggingBuildOperationProgressIntegTest extends AbstractIntegrationSpec {
             .flatten()
             .with { it as List<BuildOperationRecord.Progress> }
             .findAll { OutputEvent.isAssignableFrom(it.detailsType) }
-            // Ignore deprecations, these are checked by the testing infrastructure.
+        // Ignore deprecations, these are checked by the testing infrastructure.
             .findAll { it.details.get("category") != LoggingDeprecatedFeatureHandler.class.name }
 
         // 11 tasks + "\n" + "BUILD SUCCESSFUL" + "2 actionable tasks: 2 executed"
         // when configuration cache is enabled also "Configuration cache entry reused." and "Parallel Configuration Cache is an incubating feature."
-        def expectedEvents = GradleContextualExecuter.configCache ? 17 : 15
+        assert progressOutputEvents.size() == getNumberOfExpectedEvents()
+    }
 
-        assert progressOutputEvents.size() == expectedEvents
+    def int getNumberOfExpectedEvents() {
+        def javaWarningOffset = JavaVersion.current().majorVersion.toInteger() >= 17 ? 0 : 1
+        def configCacheOffset = GradleContextualExecuter.configCache ? 2 : 0
+        return 14 + javaWarningOffset + configCacheOffset
```

Comment:
❓ Does it make sense outside of the particular test? Perhaps `14` can be a parameter?

Reply:
I wouldn't do that for now, because there is no usage of this close by.